packages/grid/x-grid/src/XGrid.tsx

```diff
@@ -17,15 +28,29 @@ if (process.env.NODE_ENV !== 'production' && RELEASE_INFO === '__RELEASE' + '_IN
 
 LicenseInfo.setReleaseInfo(RELEASE_INFO);
 
-export type XGridProps = Omit<GridComponentProps, 'licenseStatus'>;
+export type XGridProps = GridComponentProps;
 
 const XGridRaw = React.forwardRef<HTMLDivElement, XGridProps>(function XGrid(inProps, ref) {
+  const apiRef = useGridApiRef(inProps.apiRef);
   const props = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
-  const licenseStatus = useLicenseVerifier();
 
-  return <GridComponent ref={ref} {...props} licenseStatus={licenseStatus.toString()} />;
+  useXGridComponent(apiRef, props);
+
+  return (
+    <GridContextProvider apiRef={apiRef} props={props}>
+      <GridRoot ref={ref}>
+        <GridErrorHandler>
+          <GridHeaderPlaceholder />
+          <GridBody />
+          <GridFooterPlaceholder />
+        </GridErrorHandler>
+      </GridRoot>
+    </GridContextProvider>
+  );
 });
 
+XGridRaw.defaultProps = DEFAULT_GRID_PROPS_FROM_OPTIONS;
```

Comment:
The `defaultProps` API is in its way of deprecation in React, please never use it.

```suggestion
```

Reply:
We are currently using the default props.
I do agree that it would be better not to.
Could we do it in a distinct PR to avoid making this one heavier ?

Reply:
Oh yeah, I didn't realize that we were already using defaultProps. My comment is off topic.